Problem3.rb

```diff
@@ -0,0 +1,32 @@
+#check for duplicates from from positive integers array
+include Math
+
+
+def ContainsDuplicates numbers
+
+  if numbers.nil?
+     raise ArgumentError, 'Argument is null' 
+  end 
+
+  numbers.each   do |number|
+
+    if !number.is_a? Integer
+      raise ArgumentError, 'Argument is not numeric' 
+    end
+
+    if !number.positive?
+      raise ArgumentError, 'Argument is not a positive number' 
+    end
+     
+    if numbers[number.abs-1] >0

```

Comment:
Even if the problem was correctly implemented this is not the right solution, for an input like [2, 1] you will fail by returning that the array has duplicates.

Please try to find another solution.

Reply:
I'm not sure i understand what you mean here. I added a test for this case and it seems to work. Can you give me more details of what you meant?

Reply:
Sorry my mistake, the code was hard to follow and I thought you missed an `.abs`